File: src/main/java/cz/vhromada/catalog/web/validator/ImdbMovieValidator.java
package cz.vhromada.catalog.web.validator;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

import cz.vhromada.catalog.web.fo.MovieFO;
import cz.vhromada.catalog.web.validator.constraints.Imdb;

import org.springframework.util.StringUtils;

/**
 * A class represents show validator for IMDB code constraint.
 *
 * @author Vladimir Hromada
 */
public class ImdbMovieValidator implements ConstraintValidator<Imdb, MovieFO> {

    @Override
    public void initialize(final Imdb imdb) {
    }

    @Override
    public boolean isValid(final MovieFO movie, final ConstraintValidatorContext constraintValidatorContext) {
        if (movie == null) {
            return false;
        }
        if (!movie.getImdb()) {
            return true;
        }

        return !StringUtils.isEmpty(movie.getImdbCode());
    }

}

File: src/main/kotlin/cz/vhromada/catalog/web/validator/ImdbMovieValidator.kt
package cz.vhromada.catalog.web.validator

import cz.vhromada.catalog.web.fo.MovieFO
import cz.vhromada.catalog.web.validator.constraints.Imdb
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

/**
 * A class represents show validator for IMDB code constraint.
 *
 * @author Vladimir Hromada
 */
class ImdbMovieValidator : ConstraintValidator<Imdb, MovieFO> {

    override fun isValid(movie: MovieFO?, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        if (movie == null) {
            return false
        }
        return if (!movie.imdb) {
            true
        } else !movie.imdbCode.isNullOrBlank()
    }

}
